src/internal/incompatibility.rs

```diff
@@ -155,6 +184,11 @@ impl<P: Package, VS: VersionSet> Incompatibility<P, VS> {
         if dep_term != other.get(p2) {
             return None;
         }
+        // The metadata must be identical to merge
+        let self_metadata = self.metadata();
+        if self_metadata != other.metadata() {
+            return None;
+        }
```

Comment:
I don't think this code path is needed at the moment, because we can only get here if `as_dependency` is `Some` and dependencies can't hold metadata. But perhaps they should be able to hold metadata, like what file and line number this dependency edge came from.

Thoughts?

Reply:
Removed it unless @zanieb wants to keep this

Reply:
We actually were just talking about needing that... but it can wait :D